### DISCUSSION TITLE
Install BirdNet Pi on a regular server

### DISCUSSION THREAD

**Discussion by jhackeur** [mcguirepr89/BirdNET-Pi#1262] [Q&A] [answered]
Hi,

Is it an existing way to install BirdNET-PI on a regular server instead of raspberry ?

I got this message:
"It looks like your operating system is using x86_64,
but would need to be aarch64.
Please take a look at https://birdnetwiki.pmcgui.xyz for more
information'
BirdNET-Pi requires a 64-bit OS.
It looks like your operating system is using x86_64,
but would need to be aarch64."

But i'm of course running 64-bits OS up to date.
Idea is to use my current ESX directly connected to a sound card.

thanks

**Comment by tphakala** [comment, marked as answer]
BirdNET-Go runs on x86_64 systems, https://github.com/tphakala/birdnet-go, use latest dev container image for best results for now.

docker pull ghcr.io/tphakala/birdnet-go:dev

**Reply by jhackeur** [reply]
Thanks for the message, and sorry for delay, i didn't get notificatiosn...

Unfortunately docker cannot run... Need to troubleshoot this:
"Available Capture Sources:
  0: Discard all samples (playback) or generate zero samples (capture), null
2024/11/25 08:54:35 Error selecting capture source: no suitable capture source found for device setting sysdefault"

**Reply by tphakala** [reply]
For BirdNET-Go related issues please open issue ticket at BirdNET-Go repository https://github.com/tphakala/birdnet-go

**Comment by alexbelgium** [comment]
> Hi,
> 
> Is it an existing way to install BirdNET-PI on a regular server instead of raspberry ?
> 
> I got this message: "It looks like your operating system is using x86_64, but would need to be aarch64. Please take a look at https://birdnetwiki.pmcgui.xyz for more information' BirdNET-Pi requires a 64-bit OS. It looks like your operating system is using x86_64, but would need to be aarch64."
> 
> But i'm of course running 64-bits OS up to date. Idea is to use my current ESX directly connected to a sound card.
> 
> thanks

Hi, sure, birdnet-pi was updated with support to amd64 here https://github.com/Nachtzuster/BirdNET-Pi with several updates on webui

**Reply by jhackeur** [reply]
Hi @alexbelgium ,

I cannot find the amd64 package on this link... Only for raspberry PI
Could you please give me the exact way to obtain amd64 packages ?

Thanks

**Reply by alexbelgium** [reply]
Just install it and it will work directly on amd64 ! there is no specific package ! I'm running it for the moment in an Intel CPU

**Reply by jhackeur** [reply]
ok - just tryied, but exit with error:
root@Debian-lab:/home/toto# cat installation-2024-12-04.txt
+ set -e
+ my_dir=/root/BirdNET-Pi
+ export my_dir=/root/BirdNET-Pi
+ my_dir=/root/BirdNET-Pi
+ cd /root/BirdNET-Pi/scripts
+ git log -n 1 --pretty=oneline --no-color --decorate
6d0c607e7879cd7fb9835716ef719aa3a3d6dd7e (grafted, HEAD -> main, origin/main, origin/HEAD) cleanup: propery close <pre> tag
+ source install_helpers.sh
++ uname -m
+ '[' x86_64 '!=' aarch64 ']'
++ uname -m
+ '[' x86_64 '!=' x86_64 ']'
+ ./install_config.sh
+ set -e
+ trap 'exit 1' SIGINT SIGHUP
+ echo 'Beginning ./install_config.sh'
Beginning ./install_config.sh
+ birdnet_conf=/root/BirdNET-Pi/birdnet.conf
++ curl -s4 http://ip-api.com/json
+ json=
+ exit 1


My system :
Linux Debian-lab 4.19.0-14-amd64 #1 SMP Debian 4.19.171-2 (2021-01-30) x86_64 GNU/Linux

**Reply by alexbelgium** [reply]
@Nachtzuster : any ideas? thanks!

@jhackeur : looks like this issue https://github.com/Nachtzuster/BirdNET-Pi/issues/100 that I had also encountered although the fix worked for me. I'll provide a small updated install command a bit later today to replace the offending line

**Reply by Nachtzuster** [reply]
@jhackeur correct, I think it should work now - I pushed a fix

**Reply by jhackeur** [reply]
Really better, but still an issue during the installation:
"+ case "${ARCH}-${PY_VERSION}" in
+ echo 'No tflite version found'
No tflite version found
+ WHL=
+ '[' -n '' ']'
+ pip3 install -U -r ./requirements_custom.txt
Could not open requirements file: [Errno 2] Aucun fichier ou dossier de ce type: './requirements_custom.txt'
The installation exited unsuccessfully."

File is existing in folder /root/BirdNET-Pi/

@Nachtzuster Any idea ?

**Reply by Nachtzuster** [reply]
Ancient Debian version? Please use Debian Bookworm (12)

**Reply by jhackeur** [reply]
Hi,

It was indeed a Debian version 11. I've setup a fersh install of debian 12, but still stuck..

First issue: i do not understand why you perform this check:
"if [ "$EUID" == 0 ]
  then echo "Please run as a non-root user."
  exit
fi"

It's not working for a fresh install of Debian 12, neither for Ubuntu:
Running as non root:
"~$ curl -s https://raw.githubusercontent.com/Nachtzuster/BirdNET-Pi/main/newinstaller.sh | bash
sudo: a password is required
Passwordless sudo is not working. Aborting"

Running as root:
"# curl -s https://raw.githubusercontent.com/Nachtzuster/BirdNET-Pi/main/newinstaller.sh | bash
Please run as a non-root user."

If I edit your script and remove EUID verification, script is installing without error, but webserver is not working (i got 403 refused)

Wo do we have this restriction of root ? Something blocked later is we run as root ?

**Reply by jhackeur** [reply]
ok - adding this in /etc/sudoers.d/fix the issue:
USENAME ALL=(ALL) NOPASSWD: ALL

**Reply by jhackeur** [reply]
Finally not really working as expected...

BirdNET-Pi is installed, but... All services are in inactive status:
![image](https://github.com/user-attachments/assets/ea98e1e2-2858-481d-8ef4-96cda53f807b)

and when I try to start i got an error:
![image](https://github.com/user-attachments/assets/d3a5d70e-2c21-409c-8c63-9f1cac4f64e3)


@Nachtzuster : Did you have an idea to fix that ?

**Reply by alexbelgium** [reply]
Hi, you could run manually the services installation script at `$HOME/BirdNET-Pi/scripts/install_services.sh` and it would reinstall the services. Also, you might "Tools" > "System Controls" > "Update" which will should refresh the installation in case something wrong happened in the installation.

It's really strange though the install script should have exited 1 if the services were not installed

Btw did you run the installation script as a non-root user in the end? It might be needed to appropriately set the permissions

**Reply by jhackeur** [reply]
Hi @alexbelgium 

Yes, by this change I was able to install from non-root user:
"adding this in /etc/sudoers.d/fix the issue:
USENAME ALL=(ALL) NOPASSWD: ALL"

When I try to run "install_services.sh", I got this error:
_root@debian-lab-v12:/home/toto/BirdNET-Pi$ pwd
/home/toto/BirdNET-Pi
root@debian-lab-v12:/home/toto/BirdNET-Pi$ ls -l birdnet.conf
-rw-rw-r-- 1 toto toto 9468  7 déc 14:37 birdnet.conf
toto@debian-lab-v12:~/BirdNET-Pi$ ./scripts/install_services.sh
+ trap 'rm -f ${tmpfile}' EXIT
+ trap 'exit 1' SIGINT SIGHUP
++ mktemp
+ tmpfile=/tmp/tmp.lQjpAcWeTF
+ config_file=/birdnet.conf
+ export USER=toto
+ USER=toto
+ export HOME=/home/toto
+ HOME=/home/toto
+ export PYTHON_VIRTUAL_ENV=/home/toto/BirdNET-Pi/birdnet/bin/python3
+ PYTHON_VIRTUAL_ENV=/home/toto/BirdNET-Pi/birdnet/bin/python3
+ '[' -f /birdnet.conf ']'
+ echo 'Unable to find a configuration file. Please make sure that /birdnet.conf exists.'
Unable to find a configuration file. Please make sure that /birdnet.conf exists.
+ rm -f /tmp/tmp.lQjpAcWeTF


As you can see in the ls command, birdnet.conf is existing in BirdNET-PI folder... 

EDIT: script is using variable my_dir ("config_file=$my_dir/birdnet.conf"), but this variable seems to not exist:
![image](https://github.com/user-attachments/assets/966b873e-bd3e-4c5b-a816-008e41356794)

**Reply by alexbelgium** [reply]
Mmh `config_file=/birdnet.conf` indeed is a misconfiguration, it shows that your variable `my_dir` is not set. The path in the script is `config_file=$my_dir/birdnet.conf`. You should have an env variable `my_dir=/home/$BIRDNET_USER/BirdNET-Pi/scripts`.
You could try to `export my_dir=/home/toto/BirdNET-Pi/scripts` before running the script. 

On my instance, it is an env variable ; but I notice that the install_birdnet.sh defines it before running the install_services.sh

**Reply by jhackeur** [reply]
strange indeed...

I do this (birdnet.conf is inside BirdNET-PI directly, not inside /scripts folder):
toto@debian-lab-v12:~/BirdNET-Pi$ export my_dir=/home/toto/BirdNET-Pi
toto@debian-lab-v12:~/BirdNET-Pi$ echo $my_dir
/home/toto/BirdNET-Pi


Seems better, but permissions issues:
![image](https://github.com/user-attachments/assets/7eb211af-ae3a-4ccf-9679-7272d5610274)

Do I need to set another variable for authentication ?
Or put manually my password each time ?

**Reply by alexbelgium** [reply]
Hi honestly your installation is messed up if I were you I would re-do the full installation. I have an automated build on aarch64 and amd64 that re-installs it automatically nearly weekly (as I support an [homeassistant](https://www.home-assistant.io/) addon allowing to have [BirdNET-Pi as a docker container](https://github.com/alexbelgium/hassio-addons/tree/master/birdnet-pi)), and it works straight out of the box.

To make it work, I needed (as you) to allow passwordless sudo (`echo "$USER ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers`) and set both my main user (pi) and my caddy user (caddy) as uid/gid 1000. Then I just execute `curl -s https://raw.githubusercontent.com/Nachtzuster/BirdNET-Pi/main/newinstaller.sh | bash` as my main user, from my home repository (which should be /home/toto for you) and it should install everything within /home/toto/BirdNET-Pi

Hope it will work better for you!

**Reply by jhackeur** [reply]
i will fresh install my debian again and hope it can work better.

@alexbelgium : what do you mean by "my caddy user (caddy) as uid/gid 1000" ?

**Reply by alexbelgium** [reply]
Mmh on my install it created a `caddy` user also... Not sure why as I don't see it being created in the code but at the time I didn't thought much about it and just gave it 1000:1000 uid/gid as my own user `pi` so I didn't have to think about it...

Good luck with the install! It will be worth it when it's up and running :) I look at it several times per day personally and even my non-birdy family are liking it ;)

**Reply by jhackeur** [reply]
Seems really better !
Don't know what was blocking last time... 

Waiting for better weather now to install outside :)

**Reply by jhackeur** [reply]
Thanks a lot guys

**Reply by jhackeur** [reply]
@alexbelgium still have an issue unfortunately - I open an issue here:
https://github.com/Nachtzuster/BirdNET-Pi/issues/251

If you have an idea ?
